### PR TITLE
feat(navigation): implement bottom bar navigation

### PR DESCRIPTION
**Pull request by ethanboren**
- Added BottomNavigationMenu composable
- Implemented navigation items and their click handlers
- Added tests for BottomNavigationMenu display and item clicks
- Added tests for NavigationAction.kt

### CODE REVIEW

File: app/src/main/java/com/android/solvit/MainActivity.kt [hunk 1]
```diff
@@ -9,35 +9,37 @@ import androidx.compose.material3.Surface
...
+  Text(text = "Hello, World!")
 }
-
-@Preview(showBackground = true)
+/*
 @Composable
-fun GreetingPreview() {
-  SampleAppTheme { Greeting("Android") }
-}
+fun Greeting() {
```
Comment (yshdb5) [inline app/src/main/java/com/android/solvit/MainActivity.kt:34]: The name of this function should be changed to something more clear like "SolvitApp"

**Review by yshdb5 — CHANGES_REQUESTED** [review]
### Overview

This pull request introduces the implementation of a bottom navigation bar, refactors the MainActivity, and adds necessary dependencies and tests for the new functionality.

### Positive Aspects

**Code Quality:** The code is well-structured and follows good practices for readability and maintainability.
**Modularization:** The introduction of BottomNavigationMenu and NavigationActions modularizes navigation functionality.
**Testing:** Comprehensive tests have been added for the new navigation components, ensuring reliability.
**Dependency Management:** The necessary dependencies have been added in a structured manner.

### Suggestions for Improvement

**Code Removal:** Ensure that commented-out code (e.g., in MainActivity.kt) is either removed if no longer needed or properly documented if intended for future use.
**Methods naming:** You should rename some methods like the Greeting() method in the Main Activity.

### Conclusion:

Overall, great work on implementing the bottom navigation bar and ensuring the code is well-tested and maintainable.

**Review by hjamaly — APPROVED** [review]
Everything looks good to me. I'm not sure problem with dependencies is solved but we'll manage it while merging ! Nice Work !

**Review by yshdb5 — APPROVED** [review]
### Overall

Thank you for taking into consideration our feedback. Most of the code seems good to me, we will manage the details later during development.

**Pull request merged**